ext/pgsql/pgsql.c

```diff
@@ -2576,8 +2576,9 @@ PHP_FUNCTION(pg_lo_read)
 		RETURN_THROWS();
 	}
 
-	if (ZEND_NUM_ARGS() > 1) {
-		buf_len = len < 0 ? 0 : len;
+	if (len < 0) {
+		zend_argument_value_error(2, "must be greater or equal than 0");
+		RETURN_THROWS();
```

Comment:
Not sure, but there might be some kind of "-1 for unlimited read" convention in play here.

Reply:
According to section 34.3.6. Reading Data from a Large Object from https://www.postgresql.org/docs/13/lo-interfaces.html that doesn't seem to be the case, might actually make sense to check for greater than 0.

Reply:
Actually my code is wrong, which is implies there is no test :sweat: